### PR TITLE
New algorithm to  sort pending transactions

### PR DESCRIPTION
**Pull request by ignaciopulicedonatto**
- New algorithm to sort transactions by price, but taking into account  sender and then each cluster is order by nonce.
This method first sorts list of getPendingTransactions into individual sender and sorts those lists by nonce. After the nonce ordering is satisfied, the results are merged back together by price in a heap to make it fast.

- Also added some test to check the correct behaviour of the new ordering.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/core/bc/PendingState.java [hunk 1]
```diff
@@ -91,9 +152,9 @@ public BigInteger getNonce(RskAddress addr) {
     }
 
     private void executeTransactions(Repository currentRepository, List<Transaction> pendingTransactions) {
-        pendingTransactions.stream()
-            .sorted(TRANSACTION_COMPARATOR)
-            .forEach(pendingTransaction -> executeTransaction(currentRepository, pendingTransaction));
+
```
Comment (ignaciopulicedonatto) [inline rskj-core/src/main/java/co/rsk/core/bc/PendingState.java]: I removed the old way to sort tansactions, I suppose it is okay but I wanted to just point this change to you guys pay attention .

**Comment by rskops** [comment]
SonarQube analysis reported 4 issues

1. ![MAJOR][MAJOR] [PendingState.java#L94](https://github.com/rsksmart/rskj/blob/7707cc222c727954df2632b9721cb4d067c0b3f8/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L94): The method name co.rsk.core.bc.PendingState.SortByPriceTakingIntoAccountSenderAndNonce(List) doesn't start with a lower case letter [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ANM_METHOD_NAMING_CONVENTION)
1. ![MAJOR][MAJOR] [PendingState.java#L135](https://github.com/rsksmart/rskj/blob/7707cc222c727954df2632b9721cb4d067c0b3f8/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L135): Method co.rsk.core.bc.PendingState.SortByPriceTakingIntoAccountSenderAndNonce(List) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [PendingState.java#L136](https://github.com/rsksmart/rskj/blob/7707cc222c727954df2632b9721cb4d067c0b3f8/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L136): Use isEmpty() to check whether the collection is empty or not. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1155)
1. ![MINOR][MINOR] [PendingState.java#L92](https://github.com/rsksmart/rskj/blob/7707cc222c727954df2632b9721cb4d067c0b3f8/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L92): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00100)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 3 issues

1. ![MAJOR][MAJOR] [PendingState.java#L97](https://github.com/rsksmart/rskj/blob/67100edff1e6c7e526ac401d7f26b22d982f1eea/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L97): Remove this useless assignment to local variable "mapSenderToTxs". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [PendingState.java#L125](https://github.com/rsksmart/rskj/blob/67100edff1e6c7e526ac401d7f26b22d982f1eea/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L125): Method co.rsk.core.bc.PendingState.sortByPriceTakingIntoAccountSenderAndNonce(List) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![INFO][INFO] [PendingState.java#L97](https://github.com/rsksmart/rskj/blob/67100edff1e6c7e526ac401d7f26b22d982f1eea/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L97): Dead store to mapSenderToTxs in co.rsk.core.bc.PendingState.sortByPriceTakingIntoAccountSenderAndNonce(List) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ADLS_DEAD_LOCAL_STORE)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by rskops** [comment]
SonarQube analysis reported 1 issue

1. ![MAJOR][MAJOR] [PendingState.java#L123](https://github.com/rsksmart/rskj/blob/4df0cc9c55a6e75ba2066e868c95af1d85b6fcae/rskj-core/src/main/java/co/rsk/core/bc/PendingState.java#L123): Method co.rsk.core.bc.PendingState.sortByPriceTakingIntoAccountSenderAndNonce(List) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by ignaciopulicedonatto** [comment]
Created a new pr, squashing all the commits

**Pull request closed without merging**